### PR TITLE
[Ru] New source. Yaoi-chan.me

### PR DESCRIPTION
**Pull request by Pavkazzz**

### CODE REVIEW

File: src/ru/yaoichan/src/eu/kanade/tachiyomi/extension/ru/yaoichan/Yaoichan.kt [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        var hasNextPage = false
+
+        val mangas = document.select(searchMangaSelector()).map { element ->
+            searchMangaFromElement(element)
+        }
+
+        val nextSearchPage = document.select(searchMangaNextPageSelector())
```
Comment (SnakeDoc83) [inline src/ru/yaoichan/src/eu/kanade/tachiyomi/extension/ru/yaoichan/Yaoichan.kt:157]: What are you trying to accomplish from here to line 171? It seems like an overly complicated way to get `hasNextPage`, am I not understanding something?

File: src/ru/yaoichan/src/eu/kanade/tachiyomi/extension/ru/yaoichan/Yaoichan.kt [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+        .addNetworkInterceptor(rateLimitInterceptor).build()
+
+    override fun popularMangaRequest(page: Int): Request =
+        GET("$baseUrl/mostfavorites?offset=${20 * (page - 1)}", headers)
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        var pageNum = 1
+        when {
+            page < 1 -> pageNum = 1
+            page >= 1 -> pageNum = page
```
[the remark this answers is not in the document]
Comment (Pavkazzz) [reply src/ru/yaoichan/src/eu/kanade/tachiyomi/extension/ru/yaoichan/Yaoichan.kt:45]: I've mostly copy-paste from other source, so you are right here. Refactor too: 
```
        var pageNum = when {
            page < 1 -> 1
            else -> page
        }
```

File: src/ru/yaoichan/src/eu/kanade/tachiyomi/extension/ru/yaoichan/Yaoichan.kt (hunk 1 above)
Comment (Pavkazzz) [reply src/ru/yaoichan/src/eu/kanade/tachiyomi/extension/ru/yaoichan/Yaoichan.kt:157]: Yes, it's looks like we have many way to get next page, for some reason. Pretty sure it's very rearly used part of code, will fix it if getting some real trouble with it

File: src/ru/yaoichan/src/eu/kanade/tachiyomi/extension/ru/yaoichan/Yaoichan.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (SnakeDoc83) [reply src/ru/yaoichan/src/eu/kanade/tachiyomi/extension/ru/yaoichan/Yaoichan.kt]: `page` is never less than one though, right?

File: src/ru/yaoichan/src/eu/kanade/tachiyomi/extension/ru/yaoichan/Yaoichan.kt (hunk 1 above)
Comment (SnakeDoc83) [reply src/ru/yaoichan/src/eu/kanade/tachiyomi/extension/ru/yaoichan/Yaoichan.kt:156]: I can sort of guess where you've got this from and what the original intent was, but I don't think it's needed. I just took a look at the website and I would think that you don't actually need to override `searchMangaParse()` at all, but I haven't tested.

Comment (Pavkazzz) [reply src/ru/yaoichan/src/eu/kanade/tachiyomi/extension/ru/yaoichan/Yaoichan.kt]: You are absolutely right. Remove this override

File: src/ru/yaoichan/src/eu/kanade/tachiyomi/extension/ru/yaoichan/Yaoichan.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Pavkazzz) [reply src/ru/yaoichan/src/eu/kanade/tachiyomi/extension/ru/yaoichan/Yaoichan.kt]: Yep. Remove

File: src/ru/yaoichan/src/eu/kanade/tachiyomi/extension/ru/yaoichan/Yaoichan.kt (hunk 1 above)
Comment (Pavkazzz) [reply src/ru/yaoichan/src/eu/kanade/tachiyomi/extension/ru/yaoichan/Yaoichan.kt]: Thx for review, I realy appreciate this

**Pull request merged**